bug: 토큰갱신

gimeast opened the pull request:
### 🔖 제목

-   bug: 토큰갱신

---

### 📄 본문

- next-auth 토큰 갱신이 useSession에 적용안되는 버그 수정
- 클라이언트 컴포넌트에서 사용할 fetch 훅 추가
- 서버 컴포넌트에서 사용할 커스텀 fetch util 추가
- 서버컴포넌트에서 fetch를 하는경우 특정 값을 전달하여 클라이언트 컴포넌트에서 세션업데이트 후 리페치
- 클라이언트 컴포넌트에서 fetch 결과가 401인 경우 await update()를 이용해서 세션을 업데이트 후 리페치

---

### 🔗 관련 이슈

-   관련된 이슈를 명시적으로 연결합니다.  
    예:
    -   `Closes #12`
    -   `Related to #45`

---

### ✅ 체크리스트

-   [ ] 코드가 정상적으로 동작합니다.
-   [ ] 변경 사항이 기존 기능에 영향을 주지 않습니다.
-   [ ] 리뷰어를 지정했습니다.
-   [ ] 관련 이슈를 연결했습니다.

gimeast committed:
bug: next-auth 토큰 갱신이 useSession에 적용안되는 버그 수정

- 클라이언트 컴포넌트에서 사용할 fetch 훅 추가
- 서버 컴포넌트에서 사용할 커스텀 fetch util 추가
- 서버컴포넌트에서 fetch를 하는경우 특정 값을 전달하여 클라이언트 컴포넌트에서 세션업데이트 후 리페치
- 클라이언트 컴포넌트에서 fetch 결과가 401인 경우 await update()를 이용해서 세션을 업데이트 후 리페치

gimeast committed:
Merge branch 'dev' into 164-bug-토큰갱신

# Conflicts:
#	boardgame-frontend/src/app/page.tsx
#	boardgame-frontend/src/components/board/CreateBoard.tsx

MinKyeongHyeon reviewed (approved):
수정범위가 많았을텐데 고생 많으셨어요~

gimeast committed:
bug: next-auth 토큰 갱신이 useSession에 적용안되는 버그 수정

- 클라이언트 컴포넌트에서 사용할 fetch 훅 추가
- 서버 컴포넌트에서 사용할 커스텀 fetch util 추가
- 서버컴포넌트에서 fetch를 하는경우 특정 값을 전달하여 클라이언트 컴포넌트에서 세션업데이트 후 리페치
- 클라이언트 컴포넌트에서 fetch 결과가 401인 경우 await update()를 이용해서 세션을 업데이트 후 리페치

gimeast committed:
bug: next-auth 토큰 갱신이 useSession에 적용안되는 버그 수정

- 클라이언트 컴포넌트에서 사용할 fetch 훅 추가
- 서버 컴포넌트에서 사용할 커스텀 fetch util 추가
- 서버컴포넌트에서 fetch를 하는경우 특정 값을 전달하여 클라이언트 컴포넌트에서 세션업데이트 후 리페치
- 클라이언트 컴포넌트에서 fetch 결과가 401인 경우 await update()를 이용해서 세션을 업데이트 후 리페치

jiunlee19 reviewed (approved):
고생하셨습니다!
